edge-modules/load-gen/Settings.cs

```diff
@@ -23,35 +23,33 @@ public class Settings
 
                 return new Settings(
                     configuration.GetValue("messageFrequency", TimeSpan.FromMilliseconds(20)),
-                    configuration.GetValue<double>("jitterFactor", 0.5),
                     configuration.GetValue("twinUpdateFrequency", TimeSpan.FromMilliseconds(500)),
                     configuration.GetValue<ulong>("messageSizeInBytes", 1024),
                     configuration.GetValue<TransportType>("transportType", TransportType.Amqp_Tcp_Only),
-                    configuration.GetValue<string>("outputName", "output1"));
+                    configuration.GetValue<string>("outputName", "output1"),
+                    configuration.GetValue<int>("startDelay", 1));
             });
 
         Settings(
             TimeSpan messageFrequency,
-            double jitterFactor,
             TimeSpan twinUpdateFrequency,
             ulong messageSizeInBytes,
             TransportType transportType,
-            string outputName)
+            string outputName,
+            int startDelay)
         {
             this.MessageFrequency = messageFrequency;
-            this.JitterFactor = jitterFactor;
             this.TwinUpdateFrequency = twinUpdateFrequency;
             this.MessageSizeInBytes = messageSizeInBytes;
             this.TransportType = transportType;
             this.OutputName = outputName;
+            this.StartDelay = startDelay;
```

Comment:
Recommend adding preconditions here

Reply:
Typically we only add preconditions to public methods.

Reply:
I think for this case, validation is needed for inputs since these inputs are passed directly from configuration file; which will validate the format; but not the value.  For example if startDelay is negative; this doesn't catch when getting from configuration.  We should at least validate messageSizeInBytes greater than 0; and change startDelay to uint.